### PR TITLE
Feature/aggregator bridge

### PR DESCRIPTION
**Pull request by michalsidzej**

### CODE REVIEW

File: src/store/aggregator/aggregator.selectors.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  (provider, allowTokensAddress) => {
+    if (!provider || !allowTokensAddress) {
+      return undefined;
+    }
+    const contract = AllowTokens__factory.connect(
+      allowTokensAddress,
+      provider.getSigner()
+    );
+
+    return contract as ReturnType<typeof AllowTokens__factory['connect']>;
```
Comment (meho99) [inline src/store/aggregator/aggregator.selectors.ts:80]: is this casting needed? `const contract` variable should  infer the proper  type without it

File: src/store/aggregator/aggregator.selectors.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    }
+
+    const address = tokens[startingToken].addresses[startingChain];
+
+    if (!address) {
+      return undefined;
+    }
+
+    const contract = ERC20__factory.connect(address, provider.getSigner());
+    return contract as ReturnType<typeof ERC20__factory['connect']>;
```
Comment (meho99) [inline src/store/aggregator/aggregator.selectors.ts:98]: same here, I don't think casting is needed

File: src/store/aggregator/aggregator.selectors.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    const contract = AllowTokens__factory.connect(
+      allowTokensAddress,
+      provider.getSigner()
+    );
+
+    return contract as ReturnType<typeof AllowTokens__factory['connect']>;
+  }
+);
+
+export const erc20TokenContractSelector = createSelector(
```
Comment (meho99) [inline src/store/aggregator/aggregator.selectors.ts:84]: this name sound like for some generic selector for erc20 tokens. Maybe let's change it to better describe what token is it returning? Maybe something like `startingTokenContractSelector`?

File: src/store/aggregator/aggregator.slice.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+      state.startingTokenBalance.data = payload;
+    },
+  },
+});
+
+const { actions: aggregatorActions, reducer: aggregatorReducer } =
+  aggregatorSlice;
+export { aggregatorActions, aggregatorReducer };
+
+export type AppActions = ActionsType<typeof aggregatorActions>;
```
Comment (meho99) [inline src/store/aggregator/aggregator.slice.ts:79]:
```suggestion
export type AggregatorActions = ActionsType<typeof aggregatorActions>;
```

File: src/components/TextInput/TextInput.types.ts [hunk 5]
```diff
@@ -18,6 +18,7 @@ export type ControlledInputType<FormValues> = {
   name: FieldPath<FormValues>;
   control: Control<FormValues>;
   rules?: ControllerProps['rules'];
+  showField?: boolean;
```
[the remark this answers is not in the document]
Comment (michalsidzej) [reply src/components/TextInput/TextInput.types.ts:21]: Ah, that's a leftover. It's not used anymore

**Review by meho99 — APPROVED** [review]

**Pull request merged**